src/payment/data/sagas.js

```diff
@@ -88,14 +96,39 @@ export function* handleUpdateQuantity({ payload }) {
   }
 }
 
+export function* handleSubmitPayment({ payload }) {
+  const { method, ...paymentArgs } = payload;
+  try {
+    const paymentMethodCheckout = paymentMethods[method];
+    const basket = yield select(state => ({ ...state.payment.basket }));
+    yield put(submitPayment.request());
+    const result = yield call(paymentMethodCheckout, basket, paymentArgs);
+    yield put(submitPayment.success(result));
+  } catch (error) {
+    if (error.aborted === true) {
+      // This an a user aborted action, do nothing
+    } else {
+      yield put(submitPayment.failure(error));
+      if (error.code) {
+        // Client side generated errors are simple error objects and
+        // here we wrap them into the same format as the api returns them in
+        yield call(handleErrors, { errors: [error] }, true);
```

Comment:
One thing you might want to tweak - in this app, we actually _only_ use the `messages` key for server side errors.  Robert and I had talked about removing the handling for `errors` and `field_errors` since it wasn't being used, though we didn't actually bother.  Put another way, this is actually not technically the same format as the APIs. :)